### ISSUE TITLE
[bug] PkgConfig generator is broken, when using multiple includes

### ISSUE THREAD

**Issue by derived-coder** [conan-io/conan#7056]
When using multiple includes path in 
```
def package_info(self):
    self.cpp_info.name = self.name
    self.cpp_info.libs = tools.collect_libs(self)
    self.cpp_info.includedirs = ['include', 'include/foo']
```

This will lead to a wrong generated PC file from conan.

### Environment Details (include every applicable attribute)
  * Operating System+version:Ubuntu 18.04
  * Compiler+version:gcc8
  * Conan version:1.25.1
  * Python version:3.6.9

### Logs (Executed commands with output) (Include/Attach if Applicable)

The wrong generated PC file from conan
```
prefix=/lhome/user/.conan/data2/somelib/2.13.04/company/stable/package/78f12e38896d8189fec40cf6e96af4104b967949
libdir=${prefix}/lib
includedir=${prefix}/include
includedir3=${prefix}/include/foo

Name: somelib
Description: somelib compiler
Version: 2.13.04
Libs: -L${libdir} -lthme_transport_tls  -lsomelib  -Wl,-rpath="${libdir}"
Cflags: -I${includedir} -I${includedir3}
Requires: asio openssl
```

The problem is this here:
`includedir3=${prefix}/include/somelib`

This will lead to a problem later in cmake when loading the PC file. Cmake will ignore silently the first include.

However, cmake can load the PC file correctly when following is used:
`includedir2=${prefix}/include/somelib`

I guess there is just some index too much?

**Comment by memsharded** [comment]
Hi @derived-coder 

I see that the indexing could be more intuitive, seems a small mistake there. But shouldn't be an issue, because the way it is used later:

```
Cflags: -I${includedir} -I${includedir3}
```
So the name of the local variable is irrelevant, as it is used correctly later. Can you clarify why is this a problem?

**Comment by memsharded** [comment]
Proposed https://github.com/conan-io/conan/pull/7059, to investigate this, but please let us know about the above comment

**Comment by memsharded** [comment]
Merged, fix will be in 1.26. But please still comment on the above.

**Comment by derived-coder** [comment]
Hi, here is a reproducible test case:
[test-case-pkg-config.zip](https://github.com/conan-io/conan/files/4662514/test-case-pkg-config.zip)

the actual problem was this here:
```
    def package_info(self):
        self.cpp_info.name = self.name
        self.cpp_info.libs = tools.collect_libs(self)
        self.cpp_info.includedirs = ['include', 'include/foo']

        self.env_info.HELLO_PREFIX = self.package_folder
        ## THIS IS THE PROBLEM HERE!
        self.env_info.C_INCLUDE_PATH.append(
            os.path.join(self.env_info.HELLO_PREFIX, "include"))
```
This `C_INCLUDE_PATH`, why this leads to problem later with importing the pc file via cmake, dont know...
